GUI/Types/Renderer/SimpleBoxSceneNode.cs

```diff
@@ -4,60 +4,11 @@
 
 namespace GUI.Types.Renderer
 {
-    class SimpleBoxSceneNode : SceneNode
+    class SimpleBoxSceneNode : ShapeSceneNode
     {
-        readonly Shader shader;
-        readonly int vaoHandle;
-        const int vertexCount = 8;
-
-        static readonly byte[] CubeIndices =
-        [
-#pragma warning disable format
-            0, 3, 2, 0, 1, 3, // Face1
-            4, 1, 0, 4, 5, 1, // Face2
-            4, 2, 6, 4, 0, 2, // Face3
-            6, 2, 3, 6, 3, 7, // Face4
-            5, 7, 3, 5, 3, 1, // Face5
-            4, 6, 7, 4, 7, 5, // Face6
-#pragma warning restore format
-        ];
-
         public SimpleBoxSceneNode(Scene scene, Color32 color, Vector3 scale)
-            : base(scene)
+            : base(scene, scale / -2, scale / 2, color)
         {
-            var v = scale / 2f;
-
-            LocalBoundingBox = new AABB(-v, v);
-
-            var vertices = new SimpleVertex[vertexCount];
-
-            for (var i = 0; i < vertexCount; i++)
-            {
-                var unitVector = new Vector3((i & 4) >> 2, (i & 2) >> 1, i & 1) * 2f - Vector3.One;
-                vertices[i].Position = v * unitVector;
-                vertices[i].Color = color;
-            }
-
-            // triangles: 12
-            // vertex data: 32 floats / 128 bytes
-            // indices: 9 uints / 36 bytes
-
-            shader = Scene.GuiContext.ShaderLoader.LoadShader("vrf.default");
-
-            GL.CreateVertexArrays(1, out vaoHandle);
-            GL.CreateBuffers(1, out int vboHandle);
-            GL.CreateBuffers(1, out int iboHandle);
-            GL.VertexArrayVertexBuffer(vaoHandle, 0, vboHandle, 0, SimpleVertex.SizeInBytes);
-            GL.VertexArrayElementBuffer(vaoHandle, iboHandle);
-            SimpleVertex.BindDefaultShaderLayout(vaoHandle, shader.Program);
-
-            GL.NamedBufferData(vboHandle, vertexCount * SimpleVertex.SizeInBytes, vertices, BufferUsageHint.StaticDraw);
-            GL.NamedBufferData(iboHandle, CubeIndices.Length, CubeIndices, BufferUsageHint.StaticDraw);
-
-#if DEBUG
-            var vaoLabel = nameof(SimpleBoxSceneNode);
-            GL.ObjectLabel(ObjectLabelIdentifier.VertexArray, vaoHandle, vaoLabel.Length, vaoLabel);
-#endif
         }
 
         public override void Render(Scene.RenderContext context)
```

Comment:
Perhaps ShapeSceneNode could specify whether it's transparent and switch rendering depending on that.

Reply:
Wouldn't that be a bit awkward if a child class later wanted to implement a different rendering?

Reply:
It can still override it